cmake/detect_compute_archs.cu

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 2019-2020 NVIDIA Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include <cstdio>
+#include <set>
+#include <string>
+
+int main(int argc, char** argv) {
+  std::set<std::string> archs;
+  int devices;
+  if ((cudaGetDeviceCount(&devices) == cudaSuccess) && (devices > 0)) {
+    for (int dev = 0; dev < devices; ++dev) {
+      char buff[32];
+      cudaDeviceProp prop;
+      if(cudaGetDeviceProperties(&prop, dev) != cudaSuccess) continue;
+      sprintf(buff, "%d%d", prop.major, prop.minor);
+      archs.insert(buff);
+    }
+  }
+  if (archs.empty()) {
+    printf("ALL");
```

Comment:
Doesn't look like this case is handled.

Reply:
Done.

Reply:
The case where no arches are detected and the script prints "ALL" is still not handled.